hazelcast/src/test/java/com/hazelcast/map/MapTransactionTest.java

```diff
@@ -104,6 +113,45 @@ public void run() {
     }
 
     @Test
+    public void testGetForUpdate_releasesBackupLock(){
+        Config config = new Config();
+//        config.setProperty(GroupProperties.PROP_PARTITION_COUNT, "4");
+        TestHazelcastInstanceFactory factory = createHazelcastInstanceFactory(2);
+        HazelcastInstance instance1 = factory.newHazelcastInstance(config);
+        HazelcastInstance instance2 = factory.newHazelcastInstance(config);
+        final String name = randomString();
+        final String keyOwnedByInstance2 = generateKeyOwnedBy(instance2);
+
+        instance1.executeTransaction(new TransactionalTask<Object>() {
+            @Override
+            public Object execute(TransactionalTaskContext context) throws TransactionException {
+                TransactionalMap<Object, Object> map = context.getMap(name);
+                map.getForUpdate(keyOwnedByInstance2);
+                return null;
+            }
+        });
+
+
+        Node node = TestUtil.getNode(instance1);
+        Data keyData = node.nodeEngine.toData(keyOwnedByInstance2);
+        LockService lockService = node.nodeEngine.getService(LockService.SERVICE_NAME);
+        Collection<LockResource> allLocks = lockService.getAllLocks();
+        for (LockResource lockResource : allLocks) {
+            if (keyData.equals(lockResource.getKey())){
+                assertEquals(0, lockResource.getLockCount());
+            }
+        }
+
+
+
+//        int partitionId = node.getPartitionService().getPartitionId(keyData);
+//        DefaultObjectNamespace namespace = new DefaultObjectNamespace(MapService.SERVICE_NAME, name);
+//        LockStoreImpl lockStore = lockService.getLockStore(partitionId, namespace);
+//        int lockCount = lockStore.getLockCount(keyData);
+//        assertEquals(0, lockCount);
```

Comment:
Commented out stuff here and in line 118 could be probably removed.
Afterwards we could also:
- inline `randomString()` and get rid of the name variable
- inline `lockService.getAllLocks()` and get rid of the allLocks variable

The test would be more readable that way IMHO, but these are very minor things.